src/sagemaker_training/mpi.py

```diff
@@ -341,6 +341,12 @@ def _create_command(self):
             command.extend(["-x", "FI_PROVIDER=efa"])
             # Use simple protocol to handle the out-of-order data delivery from EFA
             command.extend(["-x", "NCCL_PROTO=simple"])
+            # Further set NCCL_ALGO for modelparallel job to avoid hang, starting from torch2.0.0
+            env = environment.Environment()
+            if env.is_modelparallel_enabled:
+                import torch
+                if torch.__version__.split(".")[0] != "1":
```

Comment:
This version will use the client-side environment right? Could we get the version from the image uri or request framework version?

Reply:
I am thinking about that too. Let me work on that.